sw/device/sca/aes_serial.c

```diff
@@ -37,10 +37,45 @@ enum {
    * samples).
    */
   kIbexAesSleepCycles = 400,
+  /**
+   * Max number of encryption that can be captured with the scope
+   * 81 is selected for AES with CW Husky
+   * Note: Maybe it would be better if we use dynamic memory allocation but I
+   * am not sure whether we are supporting it or not.
+   */
+  kNumBatchOpsMax = 81,
 };
 
+/**
+ * An array of keys to be used in a batch
+ */
+uint8_t batch_keys[kNumBatchOpsMax][kAesKeyLength];
+
+/**
+ * An array of plaintexts to be used in a batch
+ */
+uint8_t batch_plaintexts[kNumBatchOpsMax][kAesTextLength];
+
+/**
+ * Key selection between fixed and random key during the batch capture.
+ */
+bool sample_fixed = true;
+
+/**
+ * Fixed key for fvsr key TVLA batch capture
+ */
+uint8_t key_fixed[kAesKeyLength];
+
 static dif_aes_t aes;
 
+dif_aes_transaction_t transaction_shared = {
```

Comment:
It looks like some functions in the file like `aes_serial_set_key()` are still using a private `dif_aes_transaction_t`. Please make sure all function use this shared transaction descriptor.